### PR TITLE
Unit tests using Travis and docker containers

### LINKED ISSUE #44: Run unit tests on every push to the repository
We need to make sure that all the unit tests pass before merging new changes. Using docker containers on travis seems like the way to go.

### PR DESCRIPTION
**Pull request by ahmadnazir**
Fixes #44

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+sudo: required
+
+language: bash
+
+services:
+  - docker
+
+before_install:
+  - docker pull penneo/mono
+  - docker run -it --rm -w '/shared' -v `pwd`:'/shared' penneo/mono /bin/sh -c "nuget install Src/Penneo/packages.config -o Src/packages; nuget install Src/PenneoTests/packages.config -o Src/packages; xbuild Src/Penneo.sln"
```
Comment (janflora) [inline .travis.yml]: Add commands to run in a shell script inside the repo.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+script:
+  - docker run -it --rm -w '/shared' -v `pwd`:'/shared' penneo/mono /bin/sh -c "mono /opt/NUnit-2.6.4/bin/nunit-console.exe Src/PenneoTests/bin/Debug/PenneoTests.dll"
```
Comment (janflora) [inline .travis.yml]: Same as above

**Comment by ahmadnazir** [comment]
@janflora Please review

**Comment by janflora** [comment]
Looks great!

**Pull request merged**